libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp

```diff
@@ -105,6 +105,36 @@ void CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element)
                 return;
             }
         }
+
+    } else if (("cpu.shares" == settingKey)) {
+        long int newValue;
+        try {
+            newValue = std::stol(settingValue);
+        } catch (std::out_of_range &err) {
+            std::string errorMessage = "The value for cpu.shares is too high";
+            log_debug() << errorMessage;
+            throw InvalidValueError(errorMessage);
+        } catch (std::invalid_argument &err) {
+            std::string errorMessage = "The value for cpu.shares is not an integer";
+            log_debug() << errorMessage;
+            throw InvalidValueError(errorMessage);
+        }
+
+        if (newValue < 2) {
```

Comment:
According to https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Resource_Management_Guide/sec-cpu.html : "The value specified in the cpu.shares file must be 2 or higher. "